src/main/resources/com/google/api/codegen/nodejs/method_sample.snip

```diff
@@ -0,0 +1,139 @@
+    @if apiMethod.hasReturnValue
+        {@methodCallSampleCodeWithReturnValue(apiMethod)};
+    @else
+        {@methodCallSampleCodeWithoutReturnValue(apiMethod)};
+    @end
+@end
+
+@private pagedOptionalArrayMethodSampleCode(apiMethod)
+    {@methodCallSampleCode(apiMethod)}.on('data', function(element) {
+        // doThingsWith(element)
+    });
+@end
+
+@private methodCallSampleCode(apiMethod)
+    api.{@apiMethod.name}(\
+        {@sampleMethodCallArgList(apiMethod.initCode.fieldSettings)})
+@end
+
+@private methodCallSampleCodeWithReturnValue(apiMethod)
+    @if sampleMethodCallArgList(apiMethod.initCode.fieldSettings)
+        api.{@apiMethod.name}(\
+            {@sampleMethodCallArgList(apiMethod.initCode.fieldSettings)}, function(err, response) {
+            if (err) {
+                console.error(err);
+                return;
+            }
+            // doThingsWith(response)
+        })
+    @else
+        api.{@apiMethod.name}(function(err, response) {
+            if (err) {
+                console.error(err);
+                return;
+            }
+            // doThingsWith(response)
+        })
+    @end
+@end
+
+@private methodCallSampleCodeWithoutReturnValue(apiMethod)
+    @if sampleMethodCallArgList(apiMethod.initCode.fieldSettings)
+        api.{@apiMethod.name}(\
+            {@sampleMethodCallArgList(apiMethod.initCode.fieldSettings)}, function(err) {
+            if (err) {
+                console.error(err);
+            }
+        })
+    @else
+        api.{@apiMethod.name}(function(err) {
+            if (err) {
+                console.error(err);
+            }
+        })
+    @end
+@end
+
+@private sampleMethodCallArgList(fieldSettings)
+    @join fieldSetting : fieldSettings on ", "
+        {@fieldSetting.identifier}
+    @end
+@end
+
+@private initCode(initCodeSpec)
+    @join line : initCodeSpec.lines
+        @switch line.lineType.toString
+        @case "StructureInitLine"
+            {@initLineStructure(line)}
+        @case "ListInitLine"
+            {@initLineList(line)}
+        @case "MapInitLine"
+            {@initLineMap(line)}
+        @case "SimpleInitLine"
+            {@initLineSimple(line)}
+        @default
+            $unhandledCase: {@line.lineType.toString}$
+        @end
+    @end
+@end
+
+@private initLineStructure(line)
+    var {@line.identifier} = {
+            @join fieldSetting : line.fieldSettings on ",".add(BREAK)
+                '{@fieldSetting.identifier}' : {@fieldSetting.identifier}
+            @end
+        };
+@end
+
+@private initLineList(line)
+    var {@line.identifier} = [{@varList(line.elementIdentifiers)}];
+@end
+
+@snippet varList(args)
+  @join arg : args on ", "
+    {@arg}
+  @end
+@end
+
+@private initLineMap(line)
+    var {@line.identifier} = {{@keyVarList(line.initEntries)}};
+@end
+
+@private keyVarList(mapEntries)
+  @join mapEntry : mapEntries vertical
+    {@mapEntry.key} : {@mapEntry.value},
+  @end
+@end
```

Comment:
be consistent with your indentation - pick 2 spaces or 4 spaces for the whole file